src/UID2.Client/UID2Encryption.cs

```diff
@@ -168,7 +167,45 @@ private static DecryptionResponse DecryptV3(byte[] encryptedId, IKeyContainer ke
             }
         }
 
-        internal static EncryptionDataResponse EncryptData(EncryptionDataRequest request, IKeyContainer keys, IdentityScope identityScope)
+        internal static EncryptionDataResponse Encrypt(string rawUid, KeyContainer keys, IdentityScope identityScope, DateTime now)
+        {
+            if (keys == null)
+            {
+                return EncryptionDataResponse.MakeError(EncryptionStatus.NotInitialized);
+            }
+            else if (!keys.IsValid(now))
+            {
+                return EncryptionDataResponse.MakeError(EncryptionStatus.KeysNotSynced);
+            }
+
+
+            if (!keys.TryGetMasterKey(now, out var masterKey))
+            {
+                return EncryptionDataResponse.MakeError(EncryptionStatus.NotAuthorizedForMasterKey);
+            }
+
+            if (!keys.TryGetDefaultKey(now, out var defaultKey))
+            {
+                return EncryptionDataResponse.MakeError(EncryptionStatus.NotAuthorizedForKey);
+            }
+
+            var expiry = now.AddSeconds(keys.TokenExpirySeconds);
+            var encryptParams = UID2TokenGenerator.DefaultParams.WithTokenExpiry(expiry);
+
+            try
+            {
+                string advertisingToken = (identityScope == IdentityScope.UID2) ? UID2TokenGenerator.GenerateUid2TokenV4(rawUid, masterKey, keys.CallerSiteId, defaultKey, encryptParams) :
+                    UID2TokenGenerator.GenerateEuidTokenV4(rawUid, masterKey, keys.CallerSiteId, defaultKey, encryptParams);
```

Comment:
actually if we set the identityScope into encryptParams directly, we don't even need GenerateEuidTokenV4 method?

Reply:
true - but doesn't matter too much as this is not public-facing